File: arm.64/langs/php/src/boot.php
<?php
require_once __DIR__ . '/find.php';
require_once __DIR__ . '/bind.php';

function boot(?string $targetDir = null): array {
    $engines = [];
    if ($envLib = getenv('PAT_LIB')) {
        if ($c = bind($envLib)) {
            $engines[] = $c;
            return $engines;
        }
    }

    $files = find($targetDir);
    foreach ($files as $file) {
        if ($c = bind($file)) {
            $engines[] = $c;
        }
    }
    return $engines;
}

File: arm.64/langs/lua/src/boot.lua
local find_mod = require("find")
local bind_mod = require("bind")

local M = {}

function M.boot(target_dir)
    local engines = {}
    local env_lib = os.getenv("PAT_LIB")
    if env_lib and env_lib ~= "" then
        local c = bind_mod.bind(env_lib)
        if c then
            table.insert(engines, c)
            return engines
        end
    end

    local files = find_mod.find(target_dir)
    for _, file in ipairs(files) do
        local c = bind_mod.bind(file)
        if c then
            table.insert(engines, c)
        end
    end
    return engines
end

return M
